### PR TITLE
tweaks to get some files to parse

### PR DESCRIPTION
**Pull request by dblodgett-cycling**
These are the little tweaks that got me going for #33 and #34 -- can add testing if you have a preference for how to do it -- I'm more familiar with testthat than tinytest.

### CODE REVIEW

File: R/dataFormatting.R [hunk 1]
```diff
@@ -188,7 +188,31 @@
...
+            x <- x[!is.na(x)]
+            if(any(x == "POSIXct")) return("POSIXct")
+            if(any(x == "character")) return("character")
+            if(any(x == "integer")) return("integer")
+            "numeric"
+          })
+          
+          message_table <- lapply(message_table, function(x) {
+            for(y in names(x)) {
+              x[[y]] <- methods::as(x[[y]], types[names(types) == y])
```
Comment (dblodgett-cycling) [inline R/dataFormatting.R:208]: realizing you'll get a warning in check if methods isn't in depends from this.

**Comment by rawmnr** [comment]
Seems to correct #32 too, thanks for the input 👍

**Comment by grimbough** [comment]
Thanks for the diagnostics and pull request.  

This was an inadvertent bug introduced by changing how the package selected the setting for reading specific datatypes.  As you noted, there was a comment about needing refactoring, and this was the case where not doing that came back to bite me as the two sections of code had diverged enough for it to be an issue.  Message that contained developer data were broken, but messages that only had standard fields were fine.  I've now done that refactoring so only function is used either both cases, rather than duplicating things.

I think the issue in #33 were actually a side effect of this.  I'm not sure it should be possible for messages of the same type to end up with different datatypes  in the same field.  The whole structure of the FIT protocol should mean that messages with the same definition have the same datatypes in the same order.  However if we were reading them wrong in the first place them that all goes out of the window.

**Comment by dblodgett-usgs** [comment]
Good deal! Thanks!

**Pull request merged**